### PR TITLE
git actions: Add push actions

### PR DESCRIPTION
**Pull request by oshoval**
For every push a latest image will be created
(overriding the previous latest unless it had a tag)
For every tag push, a tagged image will be created, and a release, which includes
a manifest corresponding to the new created tagged image.

### CODE REVIEW

File: .github/workflows/image-push.yaml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out code into the Go module directory
+        uses: actions/checkout@v2
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+
+      - name: Login to Container Registry
+        if: github.repository_owner == 'alonakaplan'
```
Comment (AlonaKaplan) [inline .github/workflows/image-push.yaml:22]: alonapaz

File: .github/workflows/image-push.yaml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        uses: docker/login-action@v1
+        with:
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Push container image
+        if: github.repository_owner == 'alonakaplan'
```
Comment (AlonaKaplan) [inline .github/workflows/image-push.yaml:30]: same

File: .github/workflows/image-push.yaml [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+      - name: Login to Container Registry
+        if: github.repository_owner == 'alonakaplan'
+        uses: docker/login-action@v1
+        with:
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Push container image
```
Comment (AlonaKaplan) [inline .github/workflows/image-push.yaml:29]: Push latest container image

File: .github/workflows/image-push.yaml [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+        if: github.repository_owner == 'alonakaplan'
+        uses: docker/build-push-action@v2
+        with:
+          context: .
+          push: true
+          tags: ghcr.io/${{ github.repository }}:latest
+          file: Dockerfile
+
+      - name: Push stable container image
```
Comment (AlonaKaplan) [inline .github/workflows/image-push.yaml:38]: Push tagged container image

File: .github/workflows/image-push.yaml (hunk 2 above)
Comment (oshoval) [reply .github/workflows/image-push.yaml:30]: it is not case sensitive
usually git url part are refereed as lower case

**Comment by oshoval** [comment]
Rebased
to solve the lint error

File: .github/workflows/image-push.yaml [hunk 5]
```diff
@@ -0,0 +1,57 @@
+name: Push container image
+on: 
+  push:
+    branches:
+      - main
+    tags:
+      - 'v*.*.*'
+env:
+  image-push-owner: 'AlonaKaplan'
```
Comment (AlonaKaplan) [inline .github/workflows/image-push.yaml:9]: Where is this `env` used?

Comment (oshoval) [reply .github/workflows/image-push.yaml:9]: Maybe something internally
I see it on other refs so copied the template
i.e
https://github.com/k8snetworkplumbingwg/whereabouts/blob/aec29ad859d08ac696b2be17ee7db398b2b0141a/.github/workflows/image-push-release.yml#L7

File: .github/workflows/image-push.yaml [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+        uses: docker/build-push-action@v2
+        with:
+          context: .
+          push: true
+          tags: ghcr.io/${{ github.repository }}:${{  github.ref_name }}
+          file: Dockerfile
+
+      - name: Update release manifests
+        if: startsWith(github.ref, 'refs/tags/')
+        run: IMAGE=ghcr.io/${{ github.repository }}:${{  github.ref_name }} hack/update-manifest.sh
```
Comment (AlonaKaplan) [inline .github/workflows/image-push.yaml:49]: what is the advantage of using ghcr and not quay as other kubevirt projects do?

Comment (oshoval) [reply .github/workflows/image-push.yaml:49]: we can login with the credentials of git itself via github actions
if we use prow for publish we can use quay, but github actions is simpler and neater
quay doesnt have any advantage

**Comment by oshoval** [comment]
https://github.com/AlonaKaplan/kubesecondarydns/actions/runs/3507917825/jobs/5875956544
ERROR: invalid tag "ghcr.io/AlonaKaplan/kubesecondarydns:latest": repository name must be lowercase
Error: buildx failed with: ERROR: invalid tag "ghcr.io/AlonaKaplan/kubesecondarydns:latest": repository name must be lowercase

**Comment by oshoval** [comment]
https://github.com/AlonaKaplan/kubesecondarydns/pull/20
fixes it

Thanks

**Pull request merged**